src/inmanta/execute/dataflow/__init__.py

```diff
@@ -161,26 +161,23 @@ class DataflowGraph:
         Graph representation of the data flow and declarative control flow of an Inmanta model.
     """
 
-    __slots__ = ("resolver", "parent", "named_nodes", "_entities", "_own_instances")
+    __slots__ = ("resolver", "parent", "_own_variables", "_own_instances")
 
     def __init__(self, resolver: "Resolver", parent: Optional["DataflowGraph"] = None) -> None:
         self.resolver: "Resolver" = resolver
         self.parent: Optional[DataflowGraph] = parent if parent is not None else None
-        self.named_nodes: Dict[str, AssignableNode] = {}
+        # keeps track of variables that have not been declared in the resolver's scope
+        self._own_variables: Dict[str, AssignableNode] = {}
         # keeps track of instance nodes and their responsible
         self._own_instances: Dict["Constructor", InstanceNode] = {}
 
-    def get_named_node(self, name: str) -> "AssignableNodeReference":
+    def get_own_variable(self, name: str) -> "AssignableNodeReference":
```

Comment:
Do I see it correctly that this can never be called in a well typed program? 
i.e. if this is called, the fault will be type fault and the datatrace is not required?

(I'm not opposed to it being here, as it already perfectly integrated, but this could be documented).

Reply:
Always a fault but not necessarily a type fault (at least, I wouldn't call it a type fault):
```
x = y
```
`y` is not defined so it will be in `own_variables`. So I think you're right that the data trace would never be required if this is called. I'll double check tomorrow.

Reply:
Double checked now, you're right. I added a comment.